apps/generator/docs/hooks.md

```diff
@@ -79,3 +84,35 @@ module.exports = {
 	};
 };
 ```
+
+## Official library
+
+It is a library of reusable hooks that you can use in your templates. You only have to add it's name to configuration: `@asyncapi/generator-hooks` and specify which hook you want to enable.
+
+This library consists of the following hooks:
+|Hook name|Hook type|Description|
+|---|---|---|
+| `createAsyncapiFile` | `generate:after` | It creates AsyncAPI file with content of the spec file passed to the generator. By default it created the file in the root of the generation output directory. This hook also supports custom parameters that user can pass to template generation. Parameter called `asyncapiFileDir` allows user to specify the location where spec file should be created. To make your template users use this parameter, you need to add it to the configuration of your template like other parameters |
+
+1. In your template configuration in `package.json` specify you want to use this library and what hook exactly:
+    ```json
+    {
+      "generator": {
+          "hooks": {
+              "@asyncapi/generator-hooks": "createAsyncapiFile"
+          }
+      }
+    }
+    ```
+1. Some hooks support custom parameters that template's user can use to specify different behaviour of the hook. To enable these, you need to also add them to the list of your template's parameters:
```

Comment:
it shows up like an error but it is all good: https://github.com/asyncapi/generator/blob/aa35277c7384a2b33767cd8706da615bdcbd1292/apps/generator/docs/hooks.md#official-library